src/main/java/org/embulk/output/sf_bulk_api/SForceTransactionalPageOutput.java

```diff
@@ -32,32 +31,34 @@ public SForceTransactionalPageOutput(ForceClient forceClient, PageReader pageRea
         this.forceClient = forceClient;
         this.pageReader = pageReader;
         this.pluginTask = pluginTask;
-        this.records = new ArrayList<>();
     }
 
     @Override
     public void add(Page page)
     {
         try {
+            ArrayList records = new ArrayList<>();
             pageReader.setPage(page);
             while (pageReader.nextRecord()) {
                 final SObject record = new SObject();
                 record.setType(this.pluginTask.getObject());
                 pageReader.getSchema().visitColumns(new SForceColumnVisitor(record, pageReader));
-                this.records.add(record);
-                if (this.records.size() >= BATCH_SIZE) {
-                    forceClient.action(records);
-                    this.records = new ArrayList<>();
+                records.add(record);
+                if (records.size() >= BATCH_SIZE) {
+                    try {
+                        forceClient.action(records);
+                    } catch (ApiFault e) {
+                        // even if some records failed to register, processing continues.
+                        logger.error(e.getExceptionCode().toString() + ":" + e.getExceptionMessage(), e);
+                    }
+                    records = new ArrayList<>();
                 }
             }
 
             if (CollectionUtils.isNotEmpty(records)) {
                 forceClient.action(records);
```

Comment:
@d-hrs 
Doesn't the ApiFault error occur here?
(because catch position is changing)

Reply:
fixed: 83fc007